frontend/src/modules/app/ToolBar.tsx

```diff
@@ -282,6 +347,42 @@ export const ToolBar = ({
           <Grid item>{StartPauseButton}</Grid>
         </Grid>
       </Grid>
+      <ModalPopup
+        withAction={true}
+        label="Set Alarms"
+        open={open}
+        onClose={handleClose}
+        onConfirm={handleConfirm}
+      >
+        <Grid container alignItems="center">
+          <Grid container alignItems="center" justify="center">
+            <Grid container alignItems="center" className={classes.marginHeader}>
+              <Grid item xs>
+                <Typography variant="h4">Keep Previous Values?</Typography>
+              </Grid>
+            </Grid>
+            <Grid item className={classes.marginContent}>
+              {alarmConfig.map((param: AlarmConfiguration) => {
+                if (alarmLimits !== null && alarmLimitsStandby !== null) {
+                  if (
+                    alarmLimits[param.stateKey].lower !==
```

Comment:
Similarly to above, shouldn't these compare `alarmLimitsRequestDraft` to `alarmLimitsRequest` rather than `alarmLimits`?

(relatedly: if the same code is being copy-pasted in 3 or 4 places, is there a way we can unify it into a reusable function, for maintainability?)

Reply:
> (relatedly: if the same code is being copy-pasted in 3 or 4 places, is there a way we can unify it into a reusable function, for maintainability?)

i think this should be fine now as I'm using a selector? I'm resolving it for now.